Testing/SystemTests/tests/qt/PythonInterfacesStartupTest.py

```diff
@@ -64,8 +66,10 @@ def _attempt_to_open_python_interface(self, interface_script):
         set_application_name(interface_script)
         # Prevents a QDialog popping up when opening certain interfaces
         set_instrument(interface_script)
+        # Some interfaces need to be closed after opening. This also ensures they process all their events
+        close_code = CLOSE_CODE_SWITCHER.get(interface_script, "")
 
         try:
-            exec(open(os.path.join(self._interface_directory, interface_script)).read())
```

Comment:
This has a real code smell to it, as were effectively echoing concatenated code into an exec. Is there no way we can call `.close()` in some sort of Qt hook when the window is closing, or am I misunderstanding what it's doing?